anymaniax/orval
# feat(fetch): support custom mutator

## Status



**READY**

## Description

follow up #1387 
I made `fetch client` available to custom mutators.
when we config to `orverride.mutator`, `fetch` client passes `url` and `params` to custom fetch function.

### default

```typescript
export const listPets = async (
  params?: ListPetsParams,
  options?: RequestInit,
): Promise<Pets> => {
  const res = await fetch(getUpdatePetsUrl(), {
    ...options,
    method: 'GET',
  });

  return res.json();
};
```

### use `orverride.mutator`

```typescript
export const listPets = async (
  params?: ListPetsParams,
  options?: RequestInit,
): Promise<Pets> => {
  return customFetch<Promise<Pets>>(getListPetsUrl(params), {
    ...options,
    method: 'GET',
  });
};
```

## Related PRs

none

## Todos


## Steps to Test or Reproduce

The sample app using custom mutator, so you can run and check it

```
cd samples/next-app-with-fetch/
yarn dev
```
